scripts/camera-controls.mjs

```diff
@@ -0,0 +1,935 @@
+        this.root.setEulerAngles(this._angles);
+    }
+
+    /**
+     * @private
+     * @param {number} dt - The delta time.
+     */
+    _smoothMove(dt) {
+        const a = dt === -1 ? 1 : lerpRate(this.moveDamping, dt);
+        this._position.lerp(this._position, this._origin, a);
+        this.root.setPosition(this._position);
+    }
+
+    /**
+     * Focus the camera on a point.
+     *
+     * @param {Vec3} point - The point.
+     * @param {Vec3} [start] - The start.
+     * @param {boolean} [smooth] - Whether to smooth the focus.
+     */
+    focus(point, start, smooth = true) {
+        if (!this._camera) {
+            return;
+        }
+        if (!start) {
+            this._origin.copy(point);
+            if (!smooth) {
+                this._position.copy(point);
+            }
+            return;
+        }
+
+        tmpV1.sub2(start, point);
+        const elev = Math.atan2(tmpV1.y, tmpV1.z) * math.RAD_TO_DEG;
+        const azim = Math.atan2(tmpV1.x, tmpV1.z) * math.RAD_TO_DEG;
+        this._dir.set(-elev, -azim);
+
+        this._origin.copy(point);
+        this._camera.entity.setPosition(start);
+        this._camera.entity.setLocalEulerAngles(0, 0, 0);
+
+        this._zoomDist = tmpV1.length();
+
+        if (!smooth) {
+            this._angles.set(this._dir.x, this._dir.y, 0);
+            this._position.copy(point);
+            this._cameraDist = this._zoomDist;
+        }
+    }
+
+    /**
+     * Reset the zoom. For orbit and panning only.
+     *
+     * @param {number} [zoomDist] - The zoom distance.
+     * @param {boolean} [smooth] - Whether to smooth the zoom.
+     */
+    resetZoom(zoomDist = 0, smooth = true) {
+        this._zoomDist = zoomDist;
+        if (!smooth) {
+            this._cameraDist = zoomDist;
+        }
+    }
+
+    /**
+     * Refocus the camera.
+     *
+     * @param {Vec3} point - The point.
+     * @param {Vec3} [start] - The start.
+     * @param {number} [zoomDist] - The zoom distance.
+     * @param {boolean} [smooth] - Whether to smooth the refocus.
+     */
+    refocus(point, start = null, zoomDist = null, smooth = true) {
+        if (zoomDist !== null) {
+            this.resetZoom(zoomDist, smooth);
+        }
+        this.focus(point, start, smooth);
+    }
+
+    /**
+     * @param {CameraComponent} camera - The camera component.
+     */
+    attach(camera) {
+        this._camera = camera;
+        this._camera.entity.setLocalEulerAngles(0, 0, 0);
+
+        window.addEventListener('wheel', this._onWheel, PASSIVE);
+        window.addEventListener('keydown', this._onKeyDown, false);
+        window.addEventListener('keyup', this._onKeyUp, false);
+        window.addEventListener('pointerdown', this._onPointerDown);
+        window.addEventListener('pointermove', this._onPointerMove);
+        window.addEventListener('pointerup', this._onPointerUp);
+        window.addEventListener('contextmenu', this._onContextMenu);
```

Comment:
I have a suspicion that you don't want to be adding listeners to the window here, at least for the pointer events. The current examples browser creates a `pc.Mouse` that is attached to `document.body`. For the Web Components examples browser, the page runs each example in an iframe and on iOS, it was not possible to use `CameraControls`. So I submitted this commit to listen on the canvas element instead:
https://github.com/playcanvas/web-components/commit/d3e83be09b7ddc512a9f758a1778bed52cf98e4b
I'm not sure if `document.body` would have worked too, but yeah, there do seem to be problems with just using `window`.

Reply:
Could be good if user was also able to override the default element.